### ISSUE TITLE
Add support for prefers-contrast in Firefox

### ISSUE THREAD

**Issue by dakahn** [carbon-design-system/carbon#6530]
In the past Carbon Components has not supported Windows High Contrast Mode on Firefox due to [lack of support for the `-ms-high-contrast` media query](https://github.com/carbon-design-system/carbon/blob/master/docs/on-windows-hcm-support.md).

[The Firefox team have PRed in a `prefers-contrast` media extension](https://bugzilla.mozilla.org/show_bug.cgi?id=1506364) that would give similar functionality (detecting when the user has enabled high contrast mode in their OS). 

I'm opening this issue to track the work toward adding Windows High Contrast mode support for Firefox v80 ([nighty build available here](https://www.mozilla.org/en-US/firefox/80.0a1/releasenotes/))

[Here's a link to the WIP Docs for this new media feature](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-contrast#Browser_compatibility)

**Comment by dakahn** [comment]
closed by https://github.com/carbon-design-system/carbon/pull/8489
